aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1700,13 +1702,47 @@ impl AptosVM {
         let mut session =
             self.new_session(resolver, SessionId::block_meta_ext(&block_metadata_ext));
 
-        let args = serialize_values(&block_metadata_ext.get_prologue_ext_move_args());
+        let block_metadata_with_randomness = match block_metadata_ext {
+            BlockMetadataExt::V0(_) => unreachable!(),
```

Comment:
why V0 exist but unreachable here?

Reply:
+1 how to ensure this is unreachable?

Reply:
i'm moving by crate by cate and this is ensured in a dependent crate.

Reply:
hmm just want to make sure it is not an attack vector. But since blockmetadata is something locally added, I think it is fine.